src/app/features/admin/components/route-form/route-form/route-form.component.ts

```diff
@@ -0,0 +1,236 @@
+import { NonNullableFormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { AlertService } from '@core/services/alert/alert.service';
+import { RouteCardComponent } from '@features/admin/components/route-card/route-card.component';
+import { ConnectedStation, Station } from '@features/admin/interfaces/station.interface';
+import { TrainRoute } from '@features/admin/interfaces/train-route.interface';
+import { RouteApiService } from '@features/admin/services/route-api/route-api.service';
+import { MIN_ROUTE_FORM_CONTROL_COUNT } from '@shared/constants/min-route-control-count';
+import { TuiButton, TuiDataList, TuiIcon, TuiLoader, TuiSurface, TuiTitle } from '@taiga-ui/core';
+import { TuiAvatar } from '@taiga-ui/kit';
+import { TuiCardLarge, TuiHeader } from '@taiga-ui/layout';
+import { TuiSelectModule, TuiTextfieldControllerModule } from '@taiga-ui/legacy';
+import { map } from 'rxjs';
+
+enum ControlsType {
+  path = 'path',
+  carriages = 'carriages',
+}
+
+@Component({
+  selector: 'dd-route-form',
+  standalone: true,
+  imports: [
+    RouteCardComponent,
+    TuiCardLarge,
+    TuiSurface,
+    TuiSelectModule,
+    TuiTextfieldControllerModule,
+    ReactiveFormsModule,
+    NgForOf,
+    TuiDataList,
+    TuiAvatar,
+    TuiButton,
+    TuiIcon,
+    TuiHeader,
+    TuiTitle,
+    TuiLoader,
+  ],
+  templateUrl: './route-form.component.html',
+  styleUrl: './route-form.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RouteFormComponent {
+  public readonly trainRoute = input<TrainRoute>();
+
+  public save = output();
+
+  private readonly alert = inject(AlertService);
+
+  private readonly routeApiService = inject(RouteApiService);
+
+  private readonly destroy = inject(DestroyRef);
+
+  private readonly fb = inject(NonNullableFormBuilder);
+
+  private readonly cdr = inject(ChangeDetectorRef);
+
+  protected readonly connectedToStations = signal<string[][]>([]);
+
+  protected readonly stations = signal<Station[]>([]);
+
+  protected readonly carriages = signal<string[]>([]);
+
+  protected readonly ControlsType = ControlsType;
+
+  private readonly allAvailableStations = computed(() =>
+    this.stations().map((station) => {
+      return {
+        id: station.id,
+      };
+    })
+  );
+
+  public form = this.fb.group({
+    path: this.fb.array([]),
+    carriages: this.fb.array([]),
+  });
+
+  public constructor() {
+    this.routeApiService
+      .getCarriages()
+      .pipe(
+        map((data) => data.map((carriage) => carriage.name)),
+        takeUntilDestroyed(this.destroy)
+      )
+      .subscribe((carriages) => {
+        this.carriages.set(carriages);
+      });
+    this.routeApiService.getStations().subscribe((data) => {
+      this.stations.set(data);
+      if (this.trainRoute()) this.updateAvailableStations(this.trainRoute()?.path || []);
+      else this.updateAvailableStations(new Array(MIN_ROUTE_FORM_CONTROL_COUNT).fill(null));
```

Comment:
Use ternary

Reply:
I can't do this because of the ESlint rule.
Should I disable this rule?
`ESLint: Expected an assignment or function call and instead saw an expression.(@typescript-eslint/no-unused-expressions)`